masonry_core/src/widgets/text_area.rs

```diff
@@ -979,8 +998,10 @@ impl<const EDITABLE: bool> Widget for TextArea<EDITABLE> {
 
     fn accessibility_role(&self) -> Role {
         if EDITABLE {
-            Role::TextInput
-            // TODO: Role::MultilineTextInput
+            match self.insert_newline {
+                InsertNewline::OnShiftEnter | InsertNewline::OnEnter => Role::MultilineTextInput,
+                _ => Role::TextInput,
+            }
```

Comment:
(this is additional to/the same thing as my previous comment)

This won't be automatically rerun when you call `set_insert_newline`, which means that until the accessibility is updated by another means, the accessibility kind will be out-of-date.